docs(router): add ignore prefix into the file naming conventions guide (#4074)
TanStack/router: docs/router/framework/react/routing/file-naming-conventions.md

| **`_` Prefix**                     | Route segments with the `_` prefix are considered to be pathless layout routes and will not be used when matching its child routes against the URL pathname.                                                                                                                                                                                                                    |
| **`_` Suffix**                     | Route segments with the `_` suffix exclude the route from being nested under any parent routes.                                                                                                                                                                                                                                                                                 |

> | **`-` Prefix**                     | Files and folders with the `-` prefix are excluded from the routing logic. They would not be added to `routeTree.gen.ts` file and can be used to colocate logic in route folders.                                                                                                                                                                                               |

Reviewer: ```suggestion
| **`-` Prefix**                     | Files and folders with the `-` prefix are excluded from the route tree. They will not be added to the `routeTree.gen.ts` file and can be used to colocate logic in route folders.                                                                                                                                                                                               |
```
